### PR TITLE
Exclude pre-release versions as per SemVer

### PR DESCRIPTION
**Pull request by LukasReschke**
As SemVer can be used apps could define a release like "10.0.0-alpha". This is something that we don't support at the moment in the server and we should filter all prereleases.

Ref https://github.com/nextcloud/server/pull/2307#issuecomment-262911588

Signed-off-by: Lukas Reschke <lukas@statuscode.ch>

@BernhardPosselt That should make you happier 😉

### CODE REVIEW

File: lib/private/App/AppStore/Fetcher/AppFetcher.php [hunk 1]
```diff
@@ -79,17 +79,18 @@ protected function fetch() {
 
 			// Filter all compatible releases
 			foreach($app['releases'] as $release) {
-				// Exclude all nightly releases
-				if($release['isNightly'] === false) {
+				// Exclude all nightly and pre-releases
```
Comment (MorrisJobke) [inline lib/private/App/AppStore/Fetcher/AppFetcher.php]: Pre-releases are defined as "contains a dash"?

Comment (LukasReschke) [inline lib/private/App/AppStore/Fetcher/AppFetcher.php]: @BernhardPosselt says so 🙈 

https://github.com/nextcloud/server/pull/2307#issuecomment-262911588

**Comment by MorrisJobke** [comment]
👍

File: lib/private/App/AppStore/Fetcher/AppFetcher.php (hunk 1 above)
Comment (BernhardPosselt) [inline lib/private/App/AppStore/Fetcher/AppFetcher.php]: Yes, check the semver spec :)

If you disallow the build metadata there's basically only versions like 1.0.0 and pre-releases like 1.0.0-alpha

**Comment by ChristophWurst** [comment]
👍

**Pull request merged**